### PR TITLE
Save users to the DB.

### PR DESCRIPTION
**Pull request by jimmyfnugent**
This is building towards users not having to log in every time. After this PR, the server should be able to handle looking up a user by session id without needing to OAuth again. Some client work needs to happen to actually request a user profile on load, though.

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -19,8 +19,14 @@ passport.use(
       consumerSecret: `${process.env.TWITTER_API_SECRET_KEY}`,
       callback: `${process.env.BASE_SERVER_URL}/oauth/twitter/callback`,
     },
-    function (token, tokenSecret, profile, cb) {
-      return cb(null, profile);
+    async function (token, tokenSecret, profile, cb) {
```
Comment (hopskipnfall) [inline server/index.js:22]: nit: use an async arrow function

File: server/src/User.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+const userSchema = new mongoose.Schema({
+  id: String,
+  twitterProfile: {
+    type: Map,
+    of: Object,
+  },
+});
+
+// Hack to prevent us from initializing the model multiple times. See
+// https://github.com/dherault/serverless-offline/issues/258.
```
Comment (hopskipnfall) [inline server/src/User.js:12]: I'm not going to click this because I don't want to know.

File: server/src/store.js [hunk 3]
```diff
@@ -67,6 +68,31 @@ export function putBit(bit) {
...
+
+export function putTwitterUser(profile) {
+  const params = {
+    id: uuid.v1(),
+    twitterProfile: profile,
+  };
+
+  return new Promise((resolve, reject) => {
+    User.create(params)
+      .then((data) => resolve(params))
```
Comment (hopskipnfall) [inline server/src/store.js:91]: do you need the params around data?

File: server/index.js (hunk 1 above)
Comment (jimmyfnugent) [reply server/index.js:22]: Done, and converted a few others as well.

File: server/src/User.js (hunk 2 above)
Comment (jimmyfnugent) [reply server/src/User.js:12]: I actually copy/pasted this from the Bit model, I assume it's the same here though

File: server/src/store.js (hunk 3 above)
Comment (jimmyfnugent) [reply server/src/store.js]: No, and in fact I just installed ESLint and configured VSCode to run it on save. So now there are a bunch more formatting changes that I'm too lazy to split into another PR 😃

**Pull request merged**